app/src/telnet.cpp

```diff
@@ -0,0 +1,822 @@
+ 	len = recv(sock, buffer, sizeof(buffer), 0); len -= 2; buffer[len] = '\0';
+	if ((len) && (len < sizeof(buffer)))
+		if (lwip_stricmp("yes", buffer) == 0)
+			OperatingParameters.Matter = true;
+		else
+			OperatingParameters.Matter = false;
+#endif
+
+#ifdef MQTT_ENABLED
+	telnet_esp32_printf ("Enable MQTT [%s]: ", (OperatingParameters.MqttEnabled) ? "Yes" : "No");
+ 	len = recv(sock, buffer, sizeof(buffer), 0); len -= 2; buffer[len] = '\0';
+	if ((len) && (len < sizeof(buffer)))
+		if (lwip_stricmp("yes", buffer) == 0)
+			OperatingParameters.MqttEnabled = true;
+		else
+			OperatingParameters.MqttEnabled = false;
+	if (OperatingParameters.MqttEnabled)
+	{
+		telnet_esp32_printf ("MQTT Broker Hostname [%s]: ", OperatingParameters.MqttBrokerHost);
+	 	len = recv(sock, buffer, sizeof(buffer), 0); len -= 2; buffer[len] = '\0';
+		if ((len) && (len < sizeof(buffer)))
+		  strcpy (OperatingParameters.MqttBrokerHost, buffer);
+
+		telnet_esp32_printf ("MQTT Broker Port [%d]: ", OperatingParameters.MqttBrokerPort);
+	 	len = recv(sock, buffer, sizeof(buffer), 0); len -= 2; buffer[len] = '\0';
+		if ((len) && (len < sizeof(buffer)))
+		  OperatingParameters.MqttBrokerPort = atoi(buffer);
+
+		telnet_esp32_printf ("MQTT Broker Username [%s]: ", OperatingParameters.MqttBrokerUsername);
+	 	len = recv(sock, buffer, sizeof(buffer), 0); len -= 2; buffer[len] = '\0';
+		if ((len) && (len < sizeof(buffer)))
+		  strcpy (OperatingParameters.MqttBrokerUsername, buffer);
+
+		telnet_esp32_printf ("MQTT Broker Password [%s]: ", OperatingParameters.MqttBrokerPassword);
+	 	len = recv(sock, buffer, sizeof(buffer), 0); len -= 2; buffer[len] = '\0';
+		if ((len) && (len < sizeof(buffer)))
+		  strcpy (OperatingParameters.MqttBrokerPassword, buffer);
+	}
+#endif
+
+	telnet_esp32_printf ("Save changes? [no]: ");
+ 	len = recv(sock, buffer, sizeof(buffer), 0); len -= 2; buffer[len] = '\0';
+	if ((len) && (len < sizeof(buffer)))
+		if (lwip_strnicmp("yes", buffer, len) == 0)
+		{
+			telnet_esp32_printf ("Resetting NVS\n");
+			// nvs_flash_init();
+			clearNVS();
+			/* Rewrite the wifi credentials after resetting NVS storage */
+			telnet_esp32_printf ("Rewriting wifi credentials\n");
+			setWifiCreds();
+			telnet_esp32_printf ("Saving thermostat config\n");
+			updateThermostatParams();
+		}
+
+	telnet_esp32_printf ("Config complete. If changes made, consider restarting (with 'Reboot')\n");
+}
+
+// All valid commands
+typedef enum {
+	HELP = 0,
+	CONFIG,
+	TEMP_UP,
+	TEMP_DOWN,
+	TEMP_SET,
+	MODE_SET,
+	MONITOR_LOG,
+	STATUS,
+	ERROR_COUNTS,
+	REBOOT,
+	QUIT,
+	UNKNOWN
+} CMD;
+
+const char *cmdStrings[] = {"HELP", "CONFIG", "UP", "DOWN", "TEMP", "MODE", "MONITOR", "STATUS", "ERROR", "REBOOT", "QUIT", NULL};
+
+#define min(x, y) ((x > y) ? y : x)
+
+static CMD lookupCommnd(char *buffer, size_t len)
+{
+	int n = 0;
+	while (cmdStrings[n] != NULL)
+	{
+		if (lwip_strnicmp(cmdStrings[n], buffer, min(strlen(cmdStrings[n]), len)) == 0)
```

Comment:
the way this string comparison is done means that an issued command like `monitorjfdklsjflsd` will be accepted as `monitor.` Not a _massive_ issue, but maybe worth fixing if this is unintentional.

Reply:
It isn't so much intentional, but I did consider this and wasn't too concerned. I felt if over time it becomes an issue, it could be addressed.

I was also picturing the number of commands may change and even the way commands are parsed may be rewritten. So I deferred this concern to later.

Do you think that's ok?

Reply:
I added a note to the MQTT module indicating more review and code cleanup was required. So this will be addressed later.